### PR TITLE
Deploy factor

### PR DESCRIPTION
**Pull request by spacefrogg**
Adds deployFactor builder to deploy factor sources as binaries.

### CODE REVIEW

**Comment by spacefrogg** [comment]
N.B. `deployFactor` depends on the updated `factor-lang`. It is, thus, added as a dependency (which should go away once #1 is merged).

File: pkgs/deployFactor/deploy-me.factor [hunk 1]
```diff
@@ -0,0 +1,27 @@
+USING: command-line io io.backend io.pathnames kernel namespaces sequences
+tools.deploy tools.deploy.config tools.deploy.backend vocabs.loader ;
+
+IN: deploy-me
+
+: load-and-deploy ( path/vocab -- )
+    normalize-path [
+        parent-directory add-vocab-root
+    ] [
+        file-name dup reload deploy
```
Comment (timor) [inline pkgs/deployFactor/deploy-me.factor]: Isn't this the same as `file-name [ reload ] [ deploy ] bi`?

File: pkgs/deployFactor/deploy-me.factor [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    load-and-deploy ;
+
+: deploy-me ( -- )
+    command-line get dup length 2 = [
+        first2 deploy-vocab
+    ] [
+        drop
+        "usage: deploy-me <PATH-TO-VOCAB> <TARGET-DIR>" print
+        nl
+    ] if ;
```
Comment (timor) [inline pkgs/deployFactor/deploy-me.factor:25]: Have a look at `if*`

Comment (spacefrogg) [reply pkgs/deployFactor/deploy-me.factor:25]: `if*` is not what I want. I don't want to keep/drop the conditional but part of what's going into it.

Comment (timor) [reply pkgs/deployFactor/deploy-me.factor:25]: ah, I see

File: default.nix [hunk 3]
```diff
@@ -97,15 +97,16 @@ in
...
+  # notmuch = super.notmuch.overrideAttrs (oldAttrs: rec {
+  #   version = "0.24.2";
+  #   name = "notmuch-${version}";
+  #   src = self.fetchurl {
+  #     url = "http://notmuchmail.org/releases/${name}.tar.gz";
+  #     sha256 = "0lfchvapk11qazdgsxj42igp9mpp83zbd0h1jj6r3ifmhikajxma";
+  #   };
+  #   doCheck = false;
+  # });
+  notmuch = callPackate ./pkgs/notmuch { };
```
Comment (spacefrogg) [inline default.nix]: You broke it! And why is this commit in this pull request?

Comment (timor) [reply default.nix]: because!

File: default.nix [hunk 4]
```diff
@@ -8,6 +8,26 @@ in
...
 {
 
+  # Usage:
+  # deployFactor "my-cool-factor-program" /factor/program/sources
+  #
+  # Output:
+  # Derivation whose build product has the following structure:
+  # /bin/<my-cool-factor-program> -> /lib/factor/<my-cool-factor-program>/<my-cool-factor-program>
+  # /lib/factor/<my-cool-factor-program>/ : Directory (may contain other used resources)
+  deployFactor = scriptName: scriptSource: super.runCommand scriptName {
```
Comment (timor) [inline default.nix:18]: @spacefrogg Why not self.runCommand?

**Pull request merged**